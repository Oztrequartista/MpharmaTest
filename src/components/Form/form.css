.form-container {
  height: 380px;
  /* border: 1px solid red; */
}

.form {
  height: 100%;
  /* border: 1px solid red; */
  /* width: 40%; */
  margin: auto;
  background: #fff;
  width: 562px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.input-container {
  /* border: 1px solid red; */
  /* width: 60%; */
  margin: auto;
  display: flex;
  flex-direction: column;
  width: 70%;
}

label {
  text-align: center;
  line-height: 2.4;
  color: #584c4c;
  font-size: 16px;
  font-weight: 600;
}

input {
  /* padding: 12px; */
  border-radius: 5px;
  border: none;
  border: 1px solid #dad9d9ea;
  background-color: #f8fbf7cf;
  padding: 16px;
}

input:focus {
  outline: 1px solid #584c4c;
}

.submit {
  /* width: 60%; */
  display: block;
  margin: 30px auto;
  padding: 15px;
  border-radius: 5px;
  border: none;

  color: #ffffff;
  font-size: 16px;
  font-weight: 600;
  text-transform: capitalize;
  text-align: center;
  cursor: pointer;
  transition: background-color 0.5s ease-in-out;
  width: 71%;
  /* 
    background-color: rgb(12, 138, 121); */
  background-color: #272121;
}

.submit:hover {
  background-color: #000000;
}

@media (max-width: 768px) {
  .form-container {
    width: 100%;
    /* border: 1px solid red; */
  }
}

@media (max-width: 468px) {
  .form-container {
    width: 350px;
    margin: auto;
  }

  .form {
    width: 100% !important;
  }

  .input-container {
    width: 80% !important;
    /* height: 60px !important; */
  }

  .submit {
    width: 80% !important;
  }
}

@media (max-width: 376px) {
  .form-container {
    width: 320px;
    margin: auto;
  }

  .form {
    width: 100% !important;
  }
}

@media (max-width: 290px) {
  .form-container {
    width: 250px;
    margin: auto;
  }

  .form {
    width: 100% !important;
  }
}

@media (max-width: 580px) {
  /* .form-container {
      width: 80% !important;
      margin: auto;
    } */

  /* .form {
      width: 100% !important;
    }
  
    .product {
      width: 100%;
    } */
}

/* @media only screen and (max-width: 500px) {
    .product-container {
      flex-direction: column;
    }
  } */
