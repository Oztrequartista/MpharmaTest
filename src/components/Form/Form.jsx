import React from "react";
import "./form.css";

const Form = ({handleFormSubmit=()=>{}, newProduct, handleInputChange=()=>{}}) => {
  const { name, itemPrice } = newProduct;
  return (
    <div className="form-container">
      <form action="" onSubmit={handleFormSubmit} className="form">
        <div className="input-container">
          <label htmlFor="Product" style={{ marginTop: "40px" }}>
            Product Name
          </label>
          <input
           placeholder="Enter A Product"
            type="text"
            name="name"
            value={name}
            onChange={handleInputChange}
            required
            onKeyDown={(e) => {
              if (e.key === "Enter") {
                handleFormSubmit();
              }
            }}
          />
        </div>

        <div className="input-container">
          <label htmlFor="Product">Price</label>
          <input
             placeholder="Enter A Price"
            type="text"
            name="itemPrice"
            value={itemPrice}
            onChange={handleInputChange}
            required
            onKeyDown={(e) => {
              if (e.key === "Enter") {
                handleFormSubmit();
              }
            }}
          />
        </div>

        <button className="submit">Add Product</button>
      </form>
    </div>
  );
};

export default Form;
