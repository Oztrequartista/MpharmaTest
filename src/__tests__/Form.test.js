import { render, screen, fireEvent } from "@testing-library/react";
import React from "react";
import Form from "../components/Form/Form";

describe("Form.js", () => {
  it("should successfully add a product", () => {
    const handleInputChange = jest.fn();
    const handleFormSubmit = jest.fn();
    const newProduct = {};
    const { container } = render(
      <Form
        handleFormSubmit={handleFormSubmit}
        handleInputChange={handleInputChange}
        newProduct={newProduct}
      />
    );
    const product = screen.getByPlaceholderText("product");
    fireEvent.change(product, {target: {value: 'Para'}});

    const price = screen.getByPlaceholderText("price");
    fireEvent.change(price, {target: {value: "23"}});

    const submit = screen.getByRole("button", {name:"Add Product"})
    fireEvent.click(submit);
    expect(handleFormSubmit).toBeCalledTimes(1);
    expect(handleInputChange).toBeCalled();


    expect(container.parentElement).toMatchSnapshot();
  
  });
});
