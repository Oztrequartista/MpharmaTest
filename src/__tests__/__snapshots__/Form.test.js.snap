// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form.js should successfully add a product 1`] = `
<body>
  <div>
    <div
      class="form-container"
    >
      <form
        action=""
        class="form"
      >
        <div
          class="input-container"
        >
          <label
            for="Product"
            style="margin-top: 40px;"
          >
            Product Name
          </label>
          <input
            name="name"
            placeholder="product"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            for="Product"
          >
            Price
          </label>
          <input
            name="itemPrice"
            placeholder="price"
            required=""
            type="text"
            value=""
          />
        </div>
        <button
          class="submit"
        >
          Add Product
        </button>
      </form>
    </div>
  </div>
</body>
`;
